packages/next/src/build/utils.ts

```diff
@@ -1953,7 +1953,7 @@ if (!process.env.NEXT_MANUAL_SIG_HANDLE) {
 }
 
 const currentPort = parseInt(process.env.PORT, 10) || 3000
-const hostname = process.env.HOSTNAME || 'localhost'
+const hostname = process.env.HOSTNAME || '127.0.0.1'
```

Comment:
I don't think this actually fixes the issue. 
When I deploy to fly.io, for instance, with a custom server, the host cannot reach the address (likely because this is localhost). 
```
WARNING The app is not listening on the expected address and will not be reachable by fly-proxy.
You can fix this by configuring your app to listen on the following addresses:
  - 0.0.0.0:8080
 ```

Reply:
This should be `0.0.0.0`, no?